test/stress/local_iterator.py

```diff
@@ -0,0 +1,236 @@
+#
+
+"""
+This script is used for PyArrowIterator performance test.
+It tracks the processing time of PyArrowIterator converting data to python objects.
+
+There are two scenarios:
+
+- row data conversion: PyArrowIterator convert data into list of tuple of python primitive objects
+- table data conversion: PyArrowIterator converts data into pyarrow table
+"""
+
+import argparse
+import base64
+import io
+import math
+import random
+import secrets
+
+import util as stress_util
+from util import task_memory_decorator, task_time_execution_decorator
+
+from snowflake.connector.arrow_context import ArrowConverterContext
+from snowflake.connector.arrow_iterator import PyArrowIterator
+from snowflake.connector.nanoarrow_arrow_iterator import (
+    PyArrowRowIterator as NanoarrowRowIterator,
+)
+from snowflake.connector.nanoarrow_arrow_iterator import (
+    PyArrowTableIterator as NanoarrowTableIterator,
+)
+from snowflake.connector.version import VERSION
+
+stress_util.print_to_console = False
+can_draw = True
+try:
+    import matplotlib.pyplot as plt
+except ImportError:
+    can_draw = False
+
+
+def remove_bytes(byte_str, num_bytes):
+    """
+    Remove a specified number of random bytes from a byte string.
+    """
+    if num_bytes >= len(byte_str):
+        return (
+            bytearray()
+        )  # Return an empty bytearray if attempting to remove more bytes than available.
+
+    indices_to_remove = random.sample(range(len(byte_str)), num_bytes)
+    new_byte_str = bytearray(
+        byte for idx, byte in enumerate(byte_str) if idx not in indices_to_remove
+    )
+    return new_byte_str
+
+
+def create_pyarrow_iterator(input_data, use_table_unit):
+    # create nanoarrow based iterator
+    return (
+        NanoarrowRowIterator(
+            None,
+            input_data,
+            ArrowConverterContext(
+                session_parameters={"TIMEZONE": "America/Los_Angeles"}
+            ),
+            False,
+            False,
+            False,
+        )
+        if not use_table_unit
+        else NanoarrowTableIterator(
+            None,
+            input_data,
+            ArrowConverterContext(
+                session_parameters={"TIMEZONE": "America/Los_Angeles"}
+            ),
+            False,
+            False,
+            False,
+        )
+    )
+
+
+def create_old_pyarrow_iterator(input_data, use_table_unit=False):
+    # created vendored arrow based iterator
+    iterator = PyArrowIterator(
+        None,
+        io.BytesIO(input_data),
+        ArrowConverterContext(session_parameters={"TIMEZONE": "America/Los_Angeles"}),
+        False,
+        False,
+        False,
+    )
+    if use_table_unit:
+        iterator.init_table_unit()
+    return iterator
+
+
+def task_for_loop_iterator(
+    input_data: bytes, create_iterator_method, use_table_unit=False
+):
+    for _ in create_iterator_method(input_data, use_table_unit):
+        pass
```

Comment:
This materializes the generator too
```suggestion
    list(create_iterator_method(input_data, use_table_unit))
```

Reply:
resolved in #1758